src/NServiceBus.Persistence.AzureStorage/Timeout/TimeoutLogic/TimeoutPersister.cs

```diff
@@ -101,9 +101,14 @@ public TimeoutPersister(string timeoutConnectionString, string timeoutDataTableN
 
             try
             {
-                await DeleteSagaEntity(timeoutId, timeoutDataTable, timeoutDataEntity).ConfigureAwait(false);
-                await DeleteTimeEntity(timeoutDataTable, timeoutDataEntity.Time.ToString(partitionKeyScope), timeoutId).ConfigureAwait(false);
-                await DeleteState(timeoutDataEntity.StateAddress).ConfigureAwait(false);
+                var deleteActions = new List<Task>(3)
```

Comment:
@dbelcham @boblangley can you weight in here? I see no reason why the deletion can't be done concurrently since the adding is also done concurrent and the three entries are separate entries.

Reply:
I don't see any harm, and the tests are passing :shipit: